compiler-cli/src/new/tests.rs

```diff
@@ -25,7 +27,8 @@ fn new() {
     assert!(path.join("test/my_project_test.gleam").exists());
     assert!(path.join(".github/workflows/test.yml").exists());
 
-    let toml = crate::fs::read(path.join("gleam.toml")).unwrap();
+    let toml =
+        crate::fs::read(Utf8PathBuf::from_path_buf(path.join("gleam.toml")).unwrap()).unwrap();
```

Comment:
Let's move this converting to utf8 path to the place where the path is created rather than here where it is used. All converting to our desired data structures should happen at the boundary of our system as much as possible.